### PR TITLE
[Refactor] Support VideoPose

### PR DESCRIPTION
**Pull request by LareinaM**
## Modification

1. Add configs for VideoPose
2. Add corresponding markdown files
3. Modify codecs to move reshaping to `PackPoseInputs`, add conversion to tensor
4. Add 2-stage pose lifter demo and corresponding doc
5. Add 3d local pose visualizer

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: configs/body_3d_keypoint/video_pose_lift/h36m/vid-pl_videopose3d-27frm-supv_8xb128-80e_h36m.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+# model settings
+model = dict(
+    type='PoseLifter',
+    backbone=dict(
+        type='TCN',
+        in_channels=2 * 17,
+        stem_channels=1024,
+        num_blocks=2,
+        kernel_sizes=(3, 3, 3),
```
Comment (Tau-J) [inline configs/body_3d_keypoint/video_pose_lift/h36m/vid-pl_videopose3d-27frm-supv_8xb128-80e_h36m.py:42]: In 1.x, `kernel_size` should be an integer. https://github.com/open-mmlab/mmpose/blob/main/mmpose/models/backbones/tcn.py

**Comment by LareinaM** [comment]
TODO: update links for downloading models

File: demo/docs/3d_human_pose_demo.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+```shell
+python  demo/body3d_pose_lifter_demo.py  \
+demo/mmdetection_cfg/faster_rcnn_r50_fpn_coco.py \
+https://download.openmmlab.com/mmdetection/v2.0/faster_rcnn/faster_rcnn_r50_fpn_1x_coco/faster_rcnn_r50_fpn_1x_coco_20200130-047c8118.pth  \
+configs/body_2d_keypoint/topdown_heatmap/coco/td-hm_hrnet-w48_8xb32-210e_coco-256x192.py \
+https://download.openmmlab.com/mmpose/top_down/hrnet/hrnet_w48_coco_256x192-b9e0b3ab_20200708.pth  \
+configs/body_3d_keypoint/video_pose_lift/h36m/vid-pl_videopose3d-243frm-supv-cpn-ft_8xb128-200e_h36m.py \
+https://download.openmmlab.com/mmpose/body3d/videopose/videopose_h36m_243frames_fullconv_supervised_cpn_ft-88f5abbb_20210527.pth  \
+--input https://user-images.githubusercontent.com/87690686/164970135-b14e424c-765a-4180-9bc8-fa8d6abc5510.mp4 \
+--out-root  vis_results  \
```
Comment (Tau-J) [inline demo/docs/3d_human_pose_demo.md:56]:
```suggestion
--output-root  vis_results  \
```

File: demo/docs/3d_human_pose_demo.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+```shell
+python  demo/body3d_pose_lifter_demo.py  \
+demo/mmdetection_cfg/faster_rcnn_r50_fpn_coco.py \
+https://download.openmmlab.com/mmdetection/v2.0/faster_rcnn/faster_rcnn_r50_fpn_1x_coco/faster_rcnn_r50_fpn_1x_coco_20200130-047c8118.pth  \
+configs/body_2d_keypoint/topdown_heatmap/posetrack18/td-hm_hrnet-w48_8xb64-20e_posetrack18-384x288.py \
+https://download.openmmlab.com/mmpose/top_down/hrnet/hrnet_w48_posetrack18_384x288-5fd6d3ff_20211130.pth  \
+configs/body_3d_keypoint/video_pose_lift/h36m/vid-pl_videopose3d-243frm-supv-cpn-ft_8xb128-200e_h36m.py \
+https://download.openmmlab.com/mmpose/body3d/videopose/videopose_h36m_243frames_fullconv_supervised_cpn_ft-88f5abbb_20210527.pth  \
+--input https://user-images.githubusercontent.com/87690686/164970135-b14e424c-765a-4180-9bc8-fa8d6abc5510.mp4 \
+--out-root  vis_results  \
```
Comment (Tau-J) [inline demo/docs/3d_human_pose_demo.md:71]:
```suggestion
--output-root  vis_results  \
```

**Review by Tau-J — APPROVED** [review]

**Pull request merged**